### PR TITLE
anthropic[patch]: multimodal

### PR DESCRIPTION
**Pull request by efriis**
- anthropic[minor]: claude 3
- x
- x

### CODE REVIEW

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 1]
```diff
@@ -36,22 +61,66 @@ def _format_messages(messages: List[BaseMessage]) -> Tuple[Optional[str], List[D
...
+            continue
+
+        role = _message_type_lookups[message.type]
+        content: Union[str, List[Dict]]
+
+        if not isinstance(message.content, str):
+            # parse as dict
+            assert isinstance(
+                message.content, list
+            ), "Anthropic message content must be str or list of dicts"
```
Comment (baskaryan) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:85]: why not accept list of str

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 2]
```diff
@@ -36,22 +61,66 @@ def _format_messages(messages: List[BaseMessage]) -> Tuple[Optional[str], List[D
...
+                    content.append(
+                        {
+                            "type": "text",
+                            "text": item,
+                        }
+                    )
+                elif isinstance(item, dict):
+                    if "type" not in item:
+                        raise ValueError("Dict content item must have a type key")
+                    if item["type"] == "image_url":
```
Comment (baskaryan) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:100]: should we support "image_url" or "image"? as an Anthropic user i might expect "image" support

Comment (efriis) [reply libs/partners/anthropic/langchain_anthropic/chat_models.py:100]: Supports both. `image_url` is the only one that triggers the conversion logic though.

File: libs/partners/anthropic/langchain_anthropic/chat_models.py (hunk 1 above)
Comment (efriis) [reply libs/partners/anthropic/langchain_anthropic/chat_models.py:85]: I think it does?

Comment (hinthornw) [reply libs/partners/anthropic/langchain_anthropic/chat_models.py:85]:
```suggestion
            ), "Anthropic message content must be str or list of strings or dicts"
```

**Comment by hinthornw** [comment]
In a followup would be nice to support image urls

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 3]
```diff
@@ -24,6 +25,30 @@
...
+      "media_type": "image/jpeg",
+      "data": "/9j/4AAQSkZJRg...",
+    }
+
+    And throws an error if it's not a b64 image
+    """
+    regex = r"^data:(?P<media_type>image/.+);base64,(?P<data>.+)$"
+    match = re.match(regex, image_url)
+    if match is None:
+        raise ValueError("Anthropic only supports base64-encoded images currently")
```
Comment (hinthornw) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:44]:
```suggestion
        raise ValueError(
            "Anthropic only supports base64-encoded images currently."
            " Example: "data:image/png;base64,'/9j/4AAQSk'..."
        )
```

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 4]
```diff
@@ -24,6 +25,30 @@
 _message_type_lookups = {"human": "user", "ai": "assistant"}
 
 
+def _format_image(image_url: str) -> Dict:
```
Comment (hinthornw) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:28]: FYI @dqbd the default anthropic format won't render properly in langsmith if implemented this way 

Think fine here

**Pull request merged**